data_safe_haven/resources/workspace/workspace.cloud_init.mustache.yaml

```diff
@@ -48,10 +20,44 @@ write_files:
           options(repos = r)
       })
 
-  - path: "/etc/skel/.xsession"
-    permissions: "0444"
+  - path: "/root/.ssh/desired_state_rsa"
+    permissions: "0600"
     content: |
-      xfce4-session
+      {{ private_key_desired_state }}
+
+  - path: "/root/desired_state.sh"
+    permissions: "0700"
+    content: |
+      #!/usr/bin/env sh
+      scp -r desired_state.{{ storage_account_data_configuration_name }}.workspace_user@desired_state.blob.core.windows.net:ansible ./
+      ansible-playbook -i localhost ansible/desired_state.yaml
+
+  - path: "/etc/systemd/system/desired-state.timer"
+    permissions: "0644"
+    content: |
+      [Unit]
+      Description=Run desired state script daily
+
+      [Timer]
+      OnCalendar=daily
+      Persistent=true
```

Comment:
Is daily often enough?

Reply:
Feels OK as a baseline. In the long term, I think we can trigger this manually.

I believe daily is midnight, plus whatever default random delay (we can specify that if we want to avoid running too much at once).

I was going to say the service is run at the end of cloud init. Then I realised I had forgotten to do that, so I've added it now 😅.